scrapy/utils/misc.py

```diff
@@ -111,3 +145,23 @@ def md5sum(file):
             break
         m.update(d)
     return m.hexdigest()
+
+
+def version_satisfied(required, provided):
+    """Return whether a provided version is greater or equal than a required
+    version.
+
+    Versions may be provided as either (x, y, z, ...) tuples or strings in the
+    form of "x.y.z...." 
+    """
+    if isinstance(required, six.string_types):
+        required = (int(x) for x in required.split('.'))
+    if isinstance(provided, six.string_types):
+        provided = (int(x) for x in provided.split('.'))
+    for r, p in six.moves.zip_longest(required, provided, fillvalue=0):
+        if p < r:
+            return False
+        elif p > r:
+            return True
+    return True
```

Comment:
I think this implementation is good enough for stable releases but I'd like better to fully support pre and post releases as well, following [PEP404](https://www.python.org/dev/peps/pep-0440/) python versioning, and preferably using something already implemented. I think we could delegate the version parsing and checking to `pkg_resources.parse_version()` in the `setuptools` package.